### PR TITLE
BREAKING CHANGES: Remove PasswordField in favor of TextField, add ema…

### PR DESCRIPTION
**Pull request by crishpeen**
…il type to TextField #DS-182

### CODE REVIEW

File: packages/web-react/src/components/TextField/useTextFieldStyleProps.tsx [hunk 1]
```diff
@@ -17,10 +16,9 @@ export interface TextFieldStyles {
 
 export function useTextFieldStyleProps(props: SpiritTextFieldProps): TextFieldStyles {
   const { validationState, isLabelHidden, ...restProps } = props;
-  const { isDisabled, isRequired, type } = restProps;
+  const { isDisabled, isRequired } = restProps;
 
-  const mainClass = `${capitalize(type)}Field`;
```
Comment (literat) [inline packages/web-react/src/components/TextField/useTextFieldStyleProps.tsx:22]: Please, remove also `capitalize` function, which is currently useless.

**Comment by literat** [comment]
@crishpeen It is a Breaking change or Feature? There are both labels in this PR.

**Review by literat — APPROVED** [review]

File: packages/web-react/src/components/TextField/README.md [hunk 2]
```diff
@@ -15,7 +16,7 @@ and show if the input is required.
 | ----------------- | ----------------------------- | ------------------------------ |
 | `id`              | string                        | Input and label identification |
 | `name`            | string                        | Input name                     |
-| `type`            | `text`, `password`            | Input type                     |
+| `type`            | `text`, `password`, `email`   | Input type                     |
```
Comment (adamkudrna) [inline packages/web-react/src/components/TextField/README.md:19]: This would maybe deserve a standalone commit. Then it could make its way into Changelog as a new feature 🙂.

**Review by adamkudrna — CHANGES_REQUESTED** [review]
Much better now! 👍🏻 

However, it seems validation states no longer work:

image

Could you please add `warning` and `success` states to the demo?

**Review by janicekt — APPROVED** [review]

File: packages/web-react/src/components/TextField/README.md (hunk 2 above)
Comment (crishpeen) [reply packages/web-react/src/components/TextField/README.md:19]: It would, but I realized that midway through doing it. :/

**Comment by crishpeen** [comment]
@literat please do the review again, as I did rebase over the main branch and had to update js&tests because of the fluid prop. I think everything is correct, but please check it again. Thanks!

File: examples/web/index.html [hunk 3]
```diff
@@ -155,6 +155,10 @@ <h2 class="typography-heading-medium-text mt-1100 mb-800">TextField</h2>
               <label for="textfield1" class="TextField__label TextField__label--required">Label</label>
               <input type="text" id="textfield1" class="TextField__input" placeholder="Placeholder" />
             </div>
+            <div class="TextField">
```
Comment (literat) [inline examples/web/index.html:158]: @crishpeen Please display all validation states like it was in PasswordField. Here in TextField is only an error state displayed. Thanks.

**Review by literat — CHANGES_REQUESTED** [review]

File: examples/web/index.html (hunk 3 above)
Comment (crishpeen) [reply examples/web/index.html:158]: Thanks, added.

**Review by literat — APPROVED** [review]

**Review by adamkudrna — APPROVED** [review]
Maybe we could present validation states together? Otherwise ✅.

obrazek
